lib/spack/spack/environment/environment.py

```diff
@@ -1247,7 +1247,10 @@ def include_concrete_envs(self):
 
     def destroy(self):
         """Remove this environment from Spack entirely."""
-        shutil.rmtree(self.path)
+        if islink(self.path):
+            os.unlink(self.path)
```

Comment:
I am torn on whether this goes here.  The distinction between managed and tracked environments seems at the moment to be a thing that wraps around the actual environment class, but this is *in* the environment class.

I think this should at least return the link target when it unlinks, so that the caller can understand when it untracked or when it destroyed, and say something useful.

Right now the env command looks like this:

```python
    for env in remove_envs:
        name = env.name
        if env.active:
            tty.die(f"Environment {name} can't be removed while activated.")
        env.destroy()
        tty.msg(f"Successfully removed environment '{name}'")
```

I think for tracked environments it do something like:

```python
        tracked_path = env.destroy()
        if tracked_path:
            tty.msg(f"Stopped tracking environment '{name}'", "env files are still in {tracked_path}")
```